satviz-consumer/src/main/java/edu/kit/satviz/consumer/gui/GuiUtils.java

```diff
@@ -147,6 +148,90 @@ public Integer fromString(String string) {
     };
   }
 
+  /**
+   * Initializes a {@code spinner} (with the initial value {@code init}) so that the user can only
+   * enter valid long integers within the given range (from {@code min} to {@code max}) while
+   * allowing the {@code spinner} to be empty if the user is currently entering a new value
+   * (i.e., has just erased an old value).
+   * <p>If the user attempts to keep the {@code spinner} empty,
+   * the {@code spinner} is reset to {@code init}.</p>
+   *
+   * @param spinner The {@code spinner} to be initialized.
+   * @param min The minimum value of the {@code spinner}.
+   * @param max The maximum value of the {@code spinner}.
+   * @param init The initial value of the {@code spinner}.
+   * @return The {@link ChangeListener} which validates that the input of the {@code spinner}
+   *         is either a valid long integer or empty.
+   */
+  public static ChangeListener<String> initializeLongSpinnerAsDouble(Spinner<Double> spinner,
+                                                                long min,
+                                                                long max,
+                                                                long init) {
+    SpinnerValueFactory<Double> spinnerValueFactory
+            = new SpinnerValueFactory.DoubleSpinnerValueFactory(min, max, init);
+
+    // prevent exception when [value is null & (enter/arrow up/arrow down) is pressed]
+    spinnerValueFactory.setConverter(new StringConverter<>() {
+      private DecimalFormat df = new DecimalFormat("#");
+
+      @Override
+      public String toString(Double object) {
+        return object == null ? "" : df.format(object);
```

Comment:
Da wir hier nur Ganzzahlen betrachten, kann man sich das DecimalFormat auch sparen und einfach direkt konvertieren.
```suggestion

      @Override
      public String toString(Double object) {
        return object == null ? "" : String.valueOf(object.longValue());
```

Reply:
Done.